core/arch/arm/tee/entry_std.c

```diff
@@ -108,14 +147,12 @@ static TEE_Result copy_in_params(const struct optee_msg_param *params,
 
 	for (n = 0; n < num_params; n++) {
 		uint32_t attr;
+		saved_attr[n] = params[n].attr;
 
-		if (params[n].attr & OPTEE_MSG_ATTR_META)
+		if (saved_attr[n] & OPTEE_MSG_ATTR_META)
 			return TEE_ERROR_BAD_PARAMETERS;
-		if (params[n].attr & OPTEE_MSG_ATTR_NONCONTIG)
-			return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
I think we should reject noncontig attributes.
Non-contig reference are used only from `register_[un]shm()` std entries to register a reference related to the parameter type `OPTEE_MSG_ATTR_RMEM_*`.
Current function `copy_params()` is called from `open/close_session()` and and `invoke_command()`. It does not expect non-contig references, only value (with.. value), tmem (with its phys addr) and rmem (with its registered ref) types.